File: src/extras-ksyntaxhighlighting/libstate.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KSyntaxHighlighting__State
#include <state.h>
#include "libstate.h"
#include "libstate.hxx"

KSyntaxHighlighting__State* KSyntaxHighlighting__State_new() {
    return new KSyntaxHighlighting::State();
}

KSyntaxHighlighting__State* KSyntaxHighlighting__State_new2(const KSyntaxHighlighting__State* other) {
    return new KSyntaxHighlighting::State(*other);
}

void KSyntaxHighlighting__State_OperatorAssign(KSyntaxHighlighting__State* self, const KSyntaxHighlighting__State* rhs) {
    self->operator=(*rhs);
}

bool KSyntaxHighlighting__State_OperatorEqual(const KSyntaxHighlighting__State* self, const KSyntaxHighlighting__State* other) {
    return (*self == *other);
}

bool KSyntaxHighlighting__State_OperatorNotEqual(const KSyntaxHighlighting__State* self, const KSyntaxHighlighting__State* other) {
    return (*self != *other);
}

bool KSyntaxHighlighting__State_IndentationBasedFoldingEnabled(const KSyntaxHighlighting__State* self) {
    return self->indentationBasedFoldingEnabled();
}

void KSyntaxHighlighting__State_Delete(KSyntaxHighlighting__State* self) {
    delete self;
}

File: src_original/extras-ksyntaxhighlighting/libstate.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/ksyntaxhighlighting-state.html
pub const ksyntaxhighlighting__state = struct {
    /// New constructs a new KSyntaxHighlighting::State object.
    ///
    ///
    pub fn New() QtC.KSyntaxHighlighting__State {
        return qtc.KSyntaxHighlighting__State_new();
    }

    /// New2 constructs a new KSyntaxHighlighting::State object.
    ///
    /// ``` other: QtC.KSyntaxHighlighting__State ```
    pub fn New2(other: ?*anyopaque) QtC.KSyntaxHighlighting__State {
        return qtc.KSyntaxHighlighting__State_new2(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-state.html#operator-eq)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__State, rhs: QtC.KSyntaxHighlighting__State ```
    pub fn OperatorAssign(self: ?*anyopaque, rhs: ?*anyopaque) void {
        qtc.KSyntaxHighlighting__State_OperatorAssign(@ptrCast(self), @ptrCast(rhs));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-state.html#operator-eq-eq)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__State, other: QtC.KSyntaxHighlighting__State ```
    pub fn OperatorEqual(self: ?*anyopaque, other: ?*anyopaque) bool {
        return qtc.KSyntaxHighlighting__State_OperatorEqual(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-state.html#operator-not-eq)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__State, other: QtC.KSyntaxHighlighting__State ```
    pub fn OperatorNotEqual(self: ?*anyopaque, other: ?*anyopaque) bool {
        return qtc.KSyntaxHighlighting__State_OperatorNotEqual(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/ksyntaxhighlighting-state.html#indentationBasedFoldingEnabled)
    ///
    /// ``` self: QtC.KSyntaxHighlighting__State ```
    pub fn IndentationBasedFoldingEnabled(self: ?*anyopaque) bool {
        return qtc.KSyntaxHighlighting__State_IndentationBasedFoldingEnabled(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KSyntaxHighlighting__State ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KSyntaxHighlighting__State_Delete(@ptrCast(self));
    }
};
